### PR TITLE
[OCPCLOUD-1436] AWS IMDSv2 machine spec validation

### PR DESCRIPTION
**Pull request by lobziik**
Enchancement: https://github.com/openshift/enhancements/pull/1075
Api changes: https://github.com/openshift/api/pull/1156
AWS machine controller changes: https://github.com/openshift/machine-api-provider-aws/pull/34

### CODE REVIEW

**Review by alexander-demichev — APPROVED** [review]
/approve

File: pkg/webhooks/machine_webhook_test.go [hunk 1]
```diff
@@ -1702,6 +1702,21 @@ func TestValidateAWSProviderSpec(t *testing.T) {
...
+			expectedOk: true,
+		},
+		{
+			testCase: "with invalid metadataServiceOptions",
+			modifySpec: func(p *machinev1.AWSMachineProviderConfig) {
+				p.MetadataServiceOptions.Authentication = "Boom"
+			},
+			expectedOk:    false,
+			expectedError: "providerSpec.metadataServiceOptions.authentication: Invalid value: \"Boom\": Allowed values either 'Optional' or 'Required'",
+		},
```
Comment (damdo) [inline pkg/webhooks/machine_webhook_test.go:1719]: We might also add a test case for the empty value?

File: pkg/webhooks/machine_webhook.go [hunk 2]
```diff
@@ -747,6 +747,20 @@ func validateAWS(m *machinev1.Machine, config *admissionConfig) (bool, []string,
...
+	switch providerSpec.MetadataServiceOptions.Authentication {
+	case "", machinev1.MetadataServiceAuthenticationOptional, machinev1.MetadataServiceAuthenticationRequired:
+		// Valid values
+	default:
+		errs = append(
+			errs,
+			field.Invalid(
+				field.NewPath("providerSpec", "metadataServiceOptions", "authentication"),
+				providerSpec.MetadataServiceOptions.Authentication,
+				fmt.Sprintf("Allowed values either '%s' or '%s'", machinev1.MetadataServiceAuthenticationOptional, machinev1.MetadataServiceAuthenticationRequired),
```
Comment (damdo) [inline pkg/webhooks/machine_webhook.go:759]: Nit:
```suggestion
				fmt.Sprintf("Allowed values are either '%s' or '%s'", machinev1.MetadataServiceAuthenticationOptional, machinev1.MetadataServiceAuthenticationRequired),
```

**Review by damdo — COMMENTED** [review]
LGTM, just a couple of minor things

File: pkg/webhooks/machine_webhook_test.go (hunk 1 above)
Comment (lobziik) [reply pkg/webhooks/machine_webhook_test.go:1719]: It is covered with other tests.

**Comment by lobziik** [comment]
I just lower down api version a bit for avoid crd updating here.
Fixed err msg.

**Review by damdo — COMMENTED** [review]
/lgtm
